libraries/resource_providers/image_streamer/api300/artifact_bundle.rb

```diff
@@ -0,0 +1,117 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require_relative '../../../resource_provider'
+
+module OneviewCookbook
+  module ImageStreamer
+    module API300
+      # DeploymentPlan Provider resource methods
+      class ArtifactBundleProvider < ResourceProvider
+        # Call the 'add_<resource_type>' methods for based on the resource_type passed in to allow using names
+        # @param [String, Symbol] resource_type Type of resource to be loaded. e.g., :GoldenImage, :FCNetwork.
+        # @param [Array] resource_array Array containing the names of the resources and whether they are readonly or not. e.g., 'DP', false
+        # @param [String] add_method String containing the name of the method which adds the arrays of resources to the artifact bundle.
+        # NOTE: Option 1: using the add_method parameter to receive which will be the add method to be used
```

Comment:
I vote for both, but with **Option 1** over **Option 2**.

I would default `add_method` to `nil`, and do the inflate if the `add_method` value is `nil`